client/src/js/filters/telephone.js

```diff
@@ -0,0 +1,60 @@
+angular.module('bhima.filters')
+.filter('telephone', TelephoneFilter);
```

Comment:
difficult to read and understand, can you comment it? so that every one will know the purpose of each line at least.